tools/cldr-code/src/main/java/org/unicode/cldr/util/DateTimeFormats.java

```diff
@@ -270,6 +270,9 @@ public DateTimeFormats set(CLDRFile file, String calendarID, boolean useStock) {
                                         + calendarID
                                         + "\"]/dateTimeFormats/availableFormats/dateFormatItem"))) {
             XPathParts parts = XPathParts.getFrozenInstance(path);
+            if ("ascii".equals(parts.findAttributeValue("dateFormatItem", "alt"))) {
```

Comment:
This doesn't allow us to build the ascii formats. I don't think it is important, but please add  javadoc note for this function, and file a ticket to (if needed!) extend the function to allow the choice of ascii or not.

Reply:
DateTimeFormats is filling in data to a DateTimePatternGenerator. The DTPG allows only one pattern per skeleton. The current behavior on main is that the pattern being populated into the DTPG is arbitrarily determined via the XPath iteration order. This change makes it such that we reliably add the non-ascii version of the pattern to the DTPG, which is the behavior before the alt=ascii was added.

It seems good to file a follow-up ticket to make this behavior configurable.

Were you okay with the PR besides this?

Reply:
Added Javadoc and filed https://unicode-org.atlassian.net/browse/CLDR-18580 as requested. PTAL